### DISCUSSION TITLE
Restricting Access to Redis via Namespace Selector

### DISCUSSION THREAD

**Discussion by torinvdb** [argoproj/argo-cd#17151] [General]
In looking at the [argocd-redis-network-policy](https://github.com/argoproj/argo-cd/blob/master/manifests/base/redis/argocd-redis-network-policy.yaml) it restricts pods based on podSelector, but not namespace which could allow an unauthorized pod labeled with the argocd labels (`argocd-server`, `argocd-repo-server`, & `argocd-application-controller`) to transit trust boundaries. 

This can be restricted by modifying the network policy or creating a new one, but may not be fully apparent to users if it's deployed via defaults. I'm not sure if this might be something that is already documented or mitigated elsewhere; seeking clarification!

**Comment by torinvdb** [comment]
Noticed this mention in the [Argo CD Vault Plugin](https://argocd-vault-plugin.readthedocs.io/en/stable/installation/) documentation, but not directly in docs related to Redis:

> The Argo CD Vault Plugin injects secrets into Kubernetes manifests generated inside the Argo CD repo-server component. Those manifests, and the secrets they contain, are cached in the Redis instance used by Argo CD. So they are available to anyone with direct access to the Redis instance. The manifests are also accessible to anyone with direct access to the repo-server.

> Mitigations: 1. Set up network policies to prevent direct access to Argo CD components (Redis and the repo-server). Make sure your cluster supports those network policies and can actually enforce them. 2. Consider running Argo CD on its own cluster, with no other applications running on it. 3. [Enable password authentication on the Redis instance](https://github.com/argoproj/argo-cd/issues/3130) (currently only supported for non-HA Argo CD installations).

**Comment by crenshaw-dev** [comment]
Hm. At a glance, I agree, I'm not sure why we don't have namespace restrictions on those namespace policies.

**Comment by torinvdb** [comment]
Do you think it'd be worth opening an issue for? I'm happy to also work on a PR as the fix would be relatively straight forward to implement a namespaceSelector and update docs so users are aware of the risks. 

Raised this in collaboration with [@smarticu5](https://github.com/smarticu5).

**Reply by crenshaw-dev** [reply]
Yes, please do!

**Comment by torinvdb** [comment]
Opened #17171 to track this issue, closing the discussion here. Thank you for your prompt reply @crenshaw-dev!
